runtime/gc_vlhgc/CardListFlushTask.cpp

```diff
@@ -119,8 +122,16 @@ MM_CardListFlushTask::run(MM_EnvironmentBase *envBase)
 						/* For Marking purposes we do not need to track references within Collection Set */
 						MM_HeapRegionDescriptorVLHGC *referencingRegion = interRegionRememberedSet->tableDescriptorForRememberedSetCard(card);
 						if (referencingRegion->containsObjects() && !referencingRegion->_markData._shouldMark) {
-							Card *cardAddress = interRegionRememberedSet->rememberedSetCardToCardAddr(env, card);
-							writeFlushToCardState(cardAddress, gmpIsActive);
+							bool needToFlush = true;
+							if (needToCheckMarkMap) {
+								void* heapAddress = convertHeapAddressFromRememberedSetCard(envBase, card);
+								uint64_t* map4Card = (uint64_t*) &((uintptr_t *)(markMap->getMarkBits()))[markMap->getSlotIndex((omrobjectptr_t) heapAddress)];
```

Comment:
this needs to go to MarkMap class, helper like anyLiveObjectsInCard(heapAddress)

Reply:
and state in a comment that implementation assumes that card covers exactly 512 bytes and heap address is card size aligned.

Reply:
you can add _compile time_ assert that BITS_PER_BYTE is 8 and CARD_SIZE_SHIFT is 9